components/resourceDetails/resourceSharing/addAgentButton/agentPickerModal/index.jsx

```diff
@@ -0,0 +1,376 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* eslint-disable react/forbid-prop-types */
+
+import React, { useContext, useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { Checkbox, createStyles } from "@material-ui/core";
+import { Button } from "@inrupt/prism-react-components";
+import { makeStyles } from "@material-ui/styles";
+import {
+  DatasetContext,
+  Table,
+  TableColumn,
+  useSession,
+} from "@inrupt/solid-ui-react";
+import { Alert } from "@material-ui/lab";
+import {
+  createSolidDataset,
+  createThing,
+  getSourceUrl,
+  getThingAll,
+  setThing,
+} from "@inrupt/solid-client";
+import { foaf } from "rdf-namespaces";
+import AccessControlContext from "../../../../../src/contexts/accessControlContext";
+import { fetchProfile } from "../../../../../src/solidClientHelpers/profile";
+import {
+  findContactInAddressBook,
+  saveContact,
+} from "../../../../../src/addressBook";
+import useAddressBook from "../../../../../src/hooks/useAddressBook";
+import useContacts from "../../../../../src/hooks/useContacts";
+import { getResourceName } from "../../../../../src/solidClientHelpers/resource";
+import PolicyHeader from "../../policyHeader";
+import AgentsTableTabs from "../../agentsTableTabs";
+import AgentsSearchBar from "../../agentsSearchBar";
+import AddAgentRow from "../addAgentRow";
+import AgentPickerEmptyState from "../agentPickerEmptyState";
+import styles from "./styles";
+import AddWebIdButton from "./addWebIdButton";
+import ConfirmationDialogContext from "../../../../../src/contexts/confirmationDialogContext";
+
+export const handleSubmit = ({
```

Comment:
So, these are the components (this and the remove button) where I was forced to do this because I cannot test the interactions with the Confirmation Dialog. I don't have any way of testing this without externalizing them, so I had no choice. I would like, however, to refactor the confirmation dialog and revisit these two components' tests when that is done.